### ISSUE TITLE
`cast storage` giving incorrect values

### ISSUE THREAD

**Issue by shuklaayush** [foundry-rs/foundry#3910]
### Component

Cast

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (18791df 2022-12-19T00:10:23.698226Z)

### What command(s) is the bug in?

cast storage

### Operating System

macOS (Apple Silicon)

### Describe the bug

Tried out the new `cast storage` command on the simple contract below and it's giving incorrect values

```solidity
// SPDX-License-Identifier: MIT

pragma solidity 0.8.17;

contract Storage {
    uint256 public n1 =  1;
    uint256 public n2 =  2;
    uint256 public n3 =  3;
    uint256 public n4 =  4;
    uint256 public n5 =  5;
    uint256 public n6 =  6;
    uint256 public n7 =  7;
    uint256 public n8 =  8;
    uint256 public n9 =  9;
    uint256 public n10 = 10;
    uint256 public n11 = 11;
    uint256 public n12 = 12;
    uint256 public n13 = 13;
    uint256 public n14 = 14;
    uint256 public n15 = 15;
    uint256 public n16 = 16;
}
```

Way to reproduce:

```bash
cast storage --rpc-url https://goerli.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161 --chain goerli 0x43C59f1BfA2829B8A661E834041fF72d4a13dd5A
```

**Comment by shuklaayush** [comment]
I think there are two issues:
1. The decimal to hex conversion here is wrong
https://github.com/foundry-rs/foundry/blob/18791dfa2f8c5ea7df99cc7bc6e00e1d20c02499/cli/src/cmd/cast/storage.rs#L200
and should probably be
```rust
            let slot_h256 = H256::from_uint(&U256::from_dec_str(&slot.slot)?);
```
2. I don't understand why the `StorageType` struct is supposed to contain a value. All `uint256` would map to the same storage type. Shouldn't the value be stored in the `Storage` struct instead?
https://github.com/foundry-rs/foundry/blob/18791dfa2f8c5ea7df99cc7bc6e00e1d20c02499/cli/src/cmd/cast/storage.rs#L207
https://github.com/gakonst/ethers-rs/blob/bb4af1c13478ab368e89be143e79b553eb956922/ethers-solc/src/artifacts/mod.rs#L1689

**Comment by DaniPopes** [comment]
You're right, first point is a bug because `U256::from_str` decodes as hex; second point I don't know, I wrote that quite a while ago lol, fixing now
